### PR TITLE
Make organization a non-null, required model field.

### PR DESCRIPTION
**Pull request by ccomb**
More tests for login and profile retrieval. Don't store null for missing organization: align with elm expecting a mandatory field that can be blank

### CODE REVIEW

File: backend/authentication/forms.py [hunk 1]
```diff
@@ -66,12 +76,13 @@ def get_token(self, user):
...
         super().save(commit)
         email = self.cleaned_data["email"]
-        for user in EmailLoginForm.get_users(self, email):
-            context = EmailLoginForm.get_mail_context(self, self.request, user)
-            EmailLoginForm.send_mail(self, email, context)
-            if "test" in sys.argv:
+        for user in MailauthEmailLoginForm.get_users(self, email):
+            context = MailauthEmailLoginForm.get_mail_context(self, self.request, user)
+            MailauthEmailLoginForm.send_mail(self, email, context)
+            if "test" in sys.argv or "backend:test" in sys.argv:
```
Comment (n1k0) [inline backend/authentication/forms.py:85]: If `test` is in `sys.argv`, it is as well in `backend:test`

File: backend/authentication/forms.py [hunk 2]
```diff
@@ -16,6 +16,16 @@ def get_login_url(self, request, token, next=None):
...
 
+    def save(self):
+        """redefine just to transmit the login_url in test mode"""
+        email = self.cleaned_data["email"]
+        for user in self.get_users(email):
+            context = self.get_mail_context(self.request, user)
+            self.send_mail(email, context)
+            if "test" in sys.argv or "backend:test" in sys.argv:
+                # hack to pass the login url to the test suite
+                os.environ["login_url"] = context["login_url"]
```
Comment (n1k0) [inline backend/authentication/forms.py:27]: Wow, this is indeed an ugly hack! Environment variable to act as a global one in production code just for tests isn't a good pattern, can we imagine another strategy here? Maybe a settings? You know you can have different django settings modules for different execution environment?

**Review by n1k0 — CHANGES_REQUESTED** [review]

File: backend/authentication/forms.py (hunk 1 above)
Comment (ccomb) [reply backend/authentication/forms.py:85]: `sys.argv` is not a string but the list of command line arguments : `"test"` is in `["python", "manage.py", "test"]` but not in `["npm", "run", "backend:test"]`

File: backend/authentication/tests.py [hunk 3]
```diff
@@ -68,17 +85,78 @@ def test_register_post(self):
...
+        # a successful login should redirect to the "next" url
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, "/")
+        # get json profile
+        response = self.client.get(reverse("profile"))
+        jsonresp = json.loads(response.content)
+        self.assertEqual(
+            list(jsonresp.keys())[:5],
+            ["email", "first_name", "last_name", "organization", "terms_of_use"],
+        )
```
Comment (n1k0) [inline backend/authentication/tests.py:159]: Why don't we test for the token here?

File: backend/authentication/forms.py (hunk 2 above)
Comment (ccomb) [reply backend/authentication/forms.py:27]: We can replace a global variable with another, anyway this code is not executed in production. Do you see a better way to pass the login_url thrown in the console to the test suite?

Comment (ccomb) [reply backend/authentication/forms.py:27]: I'm investigating the `locmem` email backend that should allow to watch in mails to be sent

Comment (n1k0) [reply backend/authentication/forms.py:27]: Now I remember there's builtin django tooling for testing email sending and contents https://docs.djangoproject.com/fr/5.0/topics/testing/tools/#django.core.mail.django.core.mail.outbox

Comment (ccomb) [reply backend/authentication/forms.py]: Yes, the `locmem` backend is automatically used in testing. I pushed a change which looks in the generated mail (I cannot look in the original context that was used to generate the mail)

**Pull request merged**